Source/WebKit/Platform/cocoa/CocoaHelpers.mm

```diff
@@ -186,6 +188,120 @@
     });
 }
 
+// MARK: NSDictionary helper methods.
+
+NSDictionary *dictionaryWithLowercaseKeys(NSDictionary *dictionary)
+{
+    if (!dictionary.count)
+        return @{ };
+
+    NSMutableDictionary *newDictionary = [NSMutableDictionary dictionaryWithCapacity:dictionary.count];
+    for (NSString *key in dictionary.allKeys) {
+        if (![key isKindOfClass:NSString.class]) {
+            ASSERT_NOT_REACHED();
+            continue;
+        }
+
+        newDictionary[key.lowercaseString] = dictionary[key];
+    }
+
+    return newDictionary;
+}
+
+NSDictionary *mergeDictionaries(NSDictionary *dictionaryA, NSDictionary *dictionaryB)
```

Comment:
`merge()` would be a fine name for this too, since this is C++ and types are part of the signature.